direct_connect.go

```diff
@@ -0,0 +1,187 @@
+			},
+		},
+		Addresses: []string{ipv4Addr.String() + "/32"},
+	}
+
+	wireguardConfig, err := peer.GetConfig(wireguardConfigArgs)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to make wireguard configuration: %w", err), 1)
+	}
+
+	// Create virtual wireguard interface with these addresses and MTU.
+	tunnelAddrs := []netip.Addr{ipv4Addr, ipv6Addr}
+	tunWireguard, tnetWireguard, err := netstack.CreateNetTUN(
+		tunnelAddrs,
+		[]netip.Addr{},
+		mtu,
+	)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to create wireguard TUN: %w", err), 1)
+	}
+	// Set the interface to promiscuous mode so forwarding works properly
+	s := tnetWireguard.Stack()
+	s.SetPromiscuousMode(1, true)
+
+	// TCP Forwarding settings.
+	catchTimeout := 5 * 1000
+	connTimeout := 5 * 1000
+	keepaliveIdle := 60
+	keepaliveCount := 3
+	keepaliveInterval := 60
+
+	// TCP Forwarding mechanism.
+	tcpConfig := tcp.Config{
+		CatchTimeout:      time.Duration(catchTimeout) * time.Millisecond,
+		ConnTimeout:       time.Duration(connTimeout) * time.Millisecond,
+		KeepaliveIdle:     time.Duration(keepaliveIdle) * time.Second,
+		KeepaliveInterval: time.Duration(keepaliveInterval) * time.Second,
+		KeepaliveCount:    int(keepaliveCount),
+		Tnet:              tnetWireguard,
+		StackLock:         &lock,
+	}
+	tcpForwarder := gtcp.NewForwarder(s, 0, 65535, tcp.Handler(tcpConfig))
+	s.SetTransportProtocolHandler(gtcp.ProtocolNumber, tcpForwarder.HandlePacket)
+
+	// UDP Forwarding mechanism.
+	udpConfig := udp.Config{
+		Tnet:      tnetWireguard,
+		StackLock: &lock,
+	}
+	s.SetTransportProtocolHandler(gudp.ProtocolNumber, udp.Handler(udpConfig))
+
+	// Configure the wireguard device with the wireguard details
+	var logger int = device.LogLevelError // Or device.LogLevelVerbose or device.LogLevelSilent
+	devWireguard := device.NewDevice(tunWireguard, conn.NewDefaultBind(), device.NewLogger(logger, ""))
+	// Configure wireguard.
+	err = devWireguard.IpcSet(wireguardConfig.AsIPC())
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to configure wireguard device: %w", err), 1)
+	}
+	err = devWireguard.Up()
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to bring up device: %w", err), 1)
+	}
+
+	// Handlers that require long-running routines:
+	// Start ICMP Handler.
+	wg.Add(1)
+	go func() {
+		icmp.Handle(tnetWireguard, &lock)
+		wg.Done()
+	}()
+
+	// Start API handler.
+	wg.Add(1)
+	go func() {
+		apiHandler(tnetWireguard, ipv4Addr, uint16(80))
+		wg.Done()
+	}()
+	log.Println("Rainforest Direct Connect running! Press Ctrl+C to exit")
+
+	wg.Wait()
+
+	return nil
+}
+
+func apiHandler(tnet *netstack.Net, addr netip.Addr, port uint16) {
+	// Stand up API server.
+	listener, err := tnet.ListenTCP(&net.TCPAddr{IP: addr.AsSlice(), Port: int(port)})
+	if err != nil {
+		log.Panic(err)
+	}
+
+	http.HandleFunc("/ping", handlePing)
+
+	err = http.Serve(listener, nil)
+	if err != nil {
+		log.Panic(err)
+	}
+}
```

Comment:
Agree it's not worth unit testing the rest, but add unit tests for `apiHandler` / `handlePing`?

Reply:
`apiHandler` is likewise difficult to test, because it needs a netstack.Net and binds/listens on a network interface (which is annoying to mock). But handlePing is simple enough, so I'll add that.

Reply:
Fair enough. Could avoid mocking and just grab a random IPv4/6 link [local address](https://en.wikipedia.org/wiki/Link-local_address)? Although I guess some hosts will not have these and would cause the tests to fail spuriously.